cache/sampleStore.js

```diff
@@ -0,0 +1,252 @@
+  fieldsToStringify: {
+    aspect: [
+      'relatedLinks',
+      'tags',
+      'criticalRange',
+      'warningRange',
+      'infoRange',
+      'okRange',
+    ],
+    sample: ['relatedLinks'],
+  },
+  indexKey: {
+    aspect: PFX + SEP + 'aspects',
+    sample: PFX + SEP + 'samples',
+    subject: PFX + SEP + 'subjects',
+  },
+  objectType: { aspect: 'aspect', sample: 'sample', subject: 'subject' },
+  prefix: PFX,
+  separator: SEP,
+};
+
+/**
+ * Generates redis key for given object type and name.
+ *
+ * @param {String} type - The object type (aspect, sample, subject).
+ * @param {String} name - The object's name or absolutePath.
+ * @returns {String} the generated redis key
+ */
+function toKey(type, name) {
+  return PFX + SEP + type + SEP + name.toLowerCase();
+} // toKey
+
+/**
+ * Clear all the "sampleStore" keys (for subjects, aspects, samples) from
+ * redis.
+ *
+ * @returns {Promise} upon completion.
+ */
+function eradicate() {
+  const promises = Object.getOwnPropertyNames(constants.indexKey)
+    .map((s) => redisClient.smembersAsync(constants.indexKey[s])
+    .then((keys) => {
+      keys.push(constants.indexKey[s]);
+      return redisClient.delAsync(keys);
+    })
+    .catch((err) => {
+      // NO-OP
+      console.error(err); // eslint-disable-line no-console
+      Promise.resolve(true);
+    }));
+  return Promise.all(promises);
+} // eradicate
+
+/**
+ * Remove nulls and stringify arrays.
+ *
+ * @param {Object} obj - The object to clean.
+ * @param {Array} arrayFields - List of array fields to stringify.
+ * @returns {Object} the object with no nulls and stringified arrays.
+ */
+function removeNullsAndStringifyArrays(obj, arrayFields) {
+  Object.keys(obj).forEach((key) => {
+    if (arrayFields.includes(key)) {
+      obj[key] = JSON.stringify(obj[key]);
+    } else if (obj[key] === null) {
+      delete obj[key];
+    }
+  });
+  return obj;
+} // removeNullsAndStringifyArrays
+
+/**
+ * Remove nulls and stringify arrays.
+ *
+ * @param {Object} a - The aspect to clean.
+ * @returns {Object} cleaned up and ready to store in redis.
+ */
+function cleanAspect(a) {
+  let retval = a.get();
+  retval = removeNullsAndStringifyArrays(retval,
+    constants.fieldsToStringify.aspect);
+  return retval;
+} // cleanAspect
+
+/**
+ * Remove the aspect field, remove nulls and stringify arrays.
+ *
+ * @param {Object} s - The sample to clean.
+ * @returns {Object} cleaned up and ready to store in redis.
+ */
+function cleanSample(s) {
+  let retval = s.get();
+  delete retval.aspect;
+  retval = removeNullsAndStringifyArrays(retval,
+    constants.fieldsToStringify.sample);
+  return retval;
+} // cleanSample
+
+/**
+ * Populate the redis sample store from the db.
+ *
+ * @returns {Promise} which resolves to the list of redis batch responses, or
+ *  false if the feature is not enabled.
+ */
+function populate() {
+  if (!featureToggles.isFeatureEnabled(constants.featureName)) {
+    return Promise.resolve(false);
+  }
```

Comment:
If it's initiated via API call (Anny's story), then it will be logged via API activity logs, but I don't see any harm in logging it. I'll add that.